libs/core/tests/unit_tests/runnables/test_runnable.py

```diff
@@ -1020,6 +1020,115 @@ def test_configurable_alts_factory() -> None:
     assert fake_llm.with_config(configurable={"llm": "chat"}).invoke("...") == "b"
 
 
+def test_configurable_fields_prefix_keys() -> None:
```

Comment:
There's a lot of stuff packed into here.. without doc-strings explaining how some of the code is supposed to work and why, i think it's hard to understand what's going on